### PR TITLE
Fix lightgbm autologging sklearn examples failure

### PR DESCRIPTION
**Pull request by BenWilson2**
Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Remove unused kwarg 'model_format' from lightgbm save call

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [X] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
Manually triggered examples with CI enablement dispatch trigger: https://github.com/BenWilson2/mlflow/actions/runs/4088026383/jobs/7049229965

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -68,12 +68,12 @@ jobs:
       - uses: ./.github/actions/setup-pyenv
 
       - name: Install dependencies
-        if: ${{ github.event_name == 'schedule' || steps.check-diff.outputs.examples_changed == 'true' }}
+        if: ${{ github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || steps.check-diff.outputs.examples_changed == 'true' }}
```
Comment (BenWilson2) [inline .github/workflows/examples.yml:71]: Added so we can manually run examples with workflow dispatch trigger.

File: mlflow/sklearn/__init__.py [hunk 2]
```diff
@@ -1333,14 +1333,23 @@ def get_input_example():
...
             model_format = get_autologging_config(flavor_name, "model_format", "xgb")
-            log_model_func(
-                self,
-                artifact_path="model",
-                signature=signature,
-                input_example=input_example,
-                registered_model_name=registered_model_name,
-                model_format=model_format,
-            )
+            if flavor_name == mlflow.xgboost.FLAVOR_NAME:
```
Comment (BenWilson2) [inline mlflow/sklearn/__init__.py:1336]: Failing examples run: https://github.com/mlflow/mlflow/actions/runs/4084422542/jobs/7041100464

**Comment by BenWilson2** [comment]
Fixes regression introduced by #7781

File: mlflow/sklearn/__init__.py [hunk 3]
```diff
@@ -1333,14 +1333,23 @@ def get_input_example():
                 flavor_name, "registered_model_name", None
             )
             model_format = get_autologging_config(flavor_name, "model_format", "xgb")
```
Comment (dbczumar) [inline mlflow/sklearn/__init__.py:1335]: Nit: can we pull this into the `if` block?

**Review by dbczumar — APPROVED** [review]
LGTM with a small nit, thanks @BenWilson2 !

Curious how we didn't catch this with non-example CI. Are we missing some test coverage? Can we add it?

File: mlflow/sklearn/__init__.py (hunk 3 above)
Comment (BenWilson2) [reply mlflow/sklearn/__init__.py:1335]: good idea :)

**Comment by BenWilson2** [comment]
@dbczumar this makes little sense to me. 
When I run the suite `tests/lightgbm/test_lightgbm_autolog.py` I get 3 expected test failures with the change introduced with that PR. 
`test_lgb_autolog_sklearn`
`test_lgb_autolog_with_sklearn_outputs_do_not_reflect_training_dataset_mutations`
`test_sklearn_api_autolog_registering_model`

Cross version testing caught it last night. Filed follow-up ticket.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**